src/Service.Tests/Unittests/MultiSourceQueryExecutionUnitTests.cs

```diff
@@ -0,0 +1,143 @@
+
+            string jsonResult2 = "{\"Name\":\"db2\"}";
+            JsonDocument document2 = JsonDocument.Parse(jsonResult2);
+            Tuple<JsonDocument, IMetadata> mockReturn2 = new(document2, PaginationMetadata.MakeEmptyPaginationMetadata());
+
+            Mock<IQueryEngine> sqlQueryEngine = new();
+            sqlQueryEngine.Setup(x => x.ExecuteAsync(It.IsAny<IMiddlewareContext>(), It.IsAny<IDictionary<string, object>>(), dataSourceName1)).Returns(Task.FromResult(mockReturn1));
+
+            Mock<IQueryEngine> cosmosQueryEngine = new();
+            cosmosQueryEngine.Setup(x => x.ExecuteAsync(It.IsAny<IMiddlewareContext>(), It.IsAny<IDictionary<string, object>>(), dataSourceName2)).Returns(Task.FromResult(mockReturn2));
+
+            Mock<IQueryEngineFactory> queryEngineFactory = new();
+            queryEngineFactory.Setup(x => x.GetQueryEngine(DatabaseType.MySQL)).Returns(sqlQueryEngine.Object);
+            queryEngineFactory.Setup(x => x.GetQueryEngine(DatabaseType.CosmosDB_NoSQL)).Returns(cosmosQueryEngine.Object);
+
+            Mock<IMutationEngineFactory> mutationEngineFactory = new();
+
+            Mock<RuntimeConfigLoader> mockLoader = new(null);
+            mockLoader.Setup(x => x.TryLoadKnownConfig(out mockConfig1, It.IsAny<bool>())).Returns(true);
+
+            RuntimeConfigProvider provider = new(mockLoader.Object);
+
+            string graphQLSchema = File.ReadAllText("MultiSourceTestSchema.gql");
+            ISchemaBuilder schemaBuilder = SchemaBuilder.New().AddDocumentFromString(graphQLSchema)
+                .AddAuthorizeDirectiveType()
+                .AddDirectiveType<ModelDirectiveType>() // Add custom directives used by DAB.
+                .AddDirectiveType<RelationshipDirectiveType>()
+                .AddDirectiveType<PrimaryKeyDirectiveType>()
+                .AddDirectiveType<DefaultValueDirectiveType>()
+                .AddDirectiveType<AutoGeneratedDirectiveType>()
+                .AddType<OrderByType>()
+                .AddType<DefaultValueType>()
+                .Use((services, next) => new ResolverMiddleware(next, queryEngineFactory.Object, mutationEngineFactory.Object, provider));
+            ISchema schema = schemaBuilder.Create();
+            IExecutionResult result = await schema.MakeExecutable().ExecuteAsync(_query);
+
+            Assert.AreEqual(1, sqlQueryEngine.Invocations.Count);
+            Assert.AreEqual(1, cosmosQueryEngine.Invocations.Count);
+
+            Assert.IsNull(result.Errors, "There should be no errors in processing of multisource query.");
+            QueryResult queryResult = (QueryResult)result;
+            Assert.IsNotNull(queryResult.Data, "Data should be returned for multisource query.");
+            IReadOnlyDictionary<string, object> data = queryResult.Data;
+            Assert.IsTrue(data.TryGetValue(queryName1, out object queryNode1));
+            Assert.IsTrue(data.TryGetValue(queryName2, out object queryNode2));
```

Comment:
`AreEqual()` and `IsTrue()` both have a message parameter that is displayed when the test fails. Could you add that to these asserts which will result in helpful error messages in the test results.

Reply:
Added error messages.